tools/privacyidea-token-janitor

```diff
@@ -509,13 +511,18 @@ def find(last_auth, assigned, active, tokeninfo_key, tokeninfo_value,
                 help='Import this PSKC file.')
 @manager.option('--preshared_key_hex', dest='preshared_key_hex',
                 help='The AES encryption key.')
-def loadtokens(pskc, preshared_key_hex):
+@manager.option('--validate_mac', dest='validate_mac',
+                help="How the file should be validated.\n"
+                     "'no_check' : Every token is parsed, ignoring HMAC\n"
+                     "'check_fail_soft' : Skip tokens with invalid HMAC\n"
+                     "'check_fail_hard' : Only import tokens if all HMAC are valid.")
+def loadtokens(pskc, preshared_key_hex, validate_mac):
     from privacyidea.lib.importotp import parsePSKCdata
 
     with open(pskc, 'r') as pskcfile:
         file_contents = pskcfile.read()
 
-    tokens = parsePSKCdata(file_contents, preshared_key_hex)
+    tokens, _ = parsePSKCdata(file_contents, preshared_key_hex, validate_mac)
     success = 0
     failed = 0
     failed_tokens = []
```

Comment:
Yes, we might add s.th. like

~~~~python
not_parsable_tokens = []
~~~~
and add simply the serial numbers to this list.

Furher below we could add

~~~~python
if not_parsable_tokens:  # Only print if tokens could not be parsed
    print("The following tokens could not be read from the PSKC file: {0!s}".format(not_parsable_tokens))
~~~~

Reply:
Should this be done instead of `failed_tokens` or additionally?

Reply:
``failed_tokens`` are the tokens, that have been successfully parsed from the xml file but could not be imported to privacyIDEA.
``not_parsable_tokens`` would already fail the parsing.
So yes, I would do this in addition.